From: Jan Glauber <jglauber@digitalocean.com>
Subject: mm-fix-shmem-thp-counters-on-migration-v3
Date: Thu, 22 Jun 2023 11:47:20 +0200

use folio_test_pmd_mappable instead of folio_test_transhuge

Link: https://lkml.kernel.org/r/20230622094720.510540-1-jglauber@digitalocean.com
Signed-off-by: Jan Glauber <jglauber@digitalocean.com>
Cc: Baolin Wang <baolin.wang@linux.alibaba.com>
Cc: "Huang, Ying" <ying.huang@intel.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/migrate.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/migrate.c~mm-fix-shmem-thp-counters-on-migration-v3
+++ a/mm/migrate.c
@@ -487,7 +487,7 @@ int folio_migrate_mapping(struct address
 			__mod_lruvec_state(old_lruvec, NR_SHMEM, -nr);
 			__mod_lruvec_state(new_lruvec, NR_SHMEM, nr);
 
-			if (folio_test_transhuge(folio)) {
+			if (folio_test_pmd_mappable(folio)) {
 				__mod_lruvec_state(old_lruvec, NR_SHMEM_THPS, -nr);
 				__mod_lruvec_state(new_lruvec, NR_SHMEM_THPS, nr);
 			}
_
